client/src/modules/Modules/CreateField.tsx

```diff
@@ -1,28 +1,69 @@
-import { useTranslation } from 'react-i18next';
-import { FC, useState } from 'react';
-import { Select, SelectContainer } from '@modules/Settings/styled/styled';
-import { typeSelectData } from '@utils/constants/Modules/typeSelectData';
+import { FC, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import { useDispatch } from 'react-redux';
+
+import { editableDataSelector } from '@redux/selectors/modules';
+import { errorAction } from '@redux/actions/error';
+import { createFieldModuleAction } from '@redux/actions/modules';
+
+import { initialfileds } from '@utils/constants/Modules/typeSelectData';
+import { useAppSelector } from '@utils/hooks/useAppSelector';
+
+import { SelectContainer } from '@modules/Settings/styled/styled';
+
 import { CreateFieldForm } from '@components/Forms/CreateFieldForm/CreateFieldForm';
 
+import { ICreateFieldProps, IFieldProps } from '@interfaces/types';
+import { ModalConfirm } from '@components/Modal/ModalConfirmSubmit/ModalConfirm';
+import { ModalButton } from '@components/Modal/ModalButton';
+import { ProtectedRoutes } from '@utils/enums/RoutesPath';
+import { modalMessageSelector, modalStatusSelector } from '@redux/selectors/modal';
+import { setModalStatusAction } from '@redux/actions/modal';
+
 export const CreateField: FC = () => {
-    const [selectValue, setSelectValue] = useState('checkbox');
-    const { t } = useTranslation();
+    const [currentField, setCurrentField] = useState<IFieldProps>();
```

Comment:
why you don't pass default useState value? Setting in useEffect will cause rerender